src/dscanner/analysis/incorrect_infinite_range.d

```diff
@@ -13,85 +13,84 @@ import dparse.lexer;
+	extern(D) this(string fileName)
 	{
-		super(fileName, null, skipTests);
+		super(fileName);
 	}
 
-	override void visit(const StructBody structBody)
+	override void visit(AST.StructDeclaration sd)
 	{
-		inStruct++;
-		structBody.accept(this);
-		inStruct--;
+		inAggregate++;
+		super.visit(sd);
+		inAggregate--;
 	}
 
-	override void visit(const FunctionDeclaration fd)
+	override void visit(AST.ClassDeclaration cd)
 	{
-		if (inStruct > 0 && fd.name.text == "empty")
-		{
-			line = fd.name.line;
-			column = fd.name.column;
-			fd.accept(this);
-		}
+		inAggregate++;
+		super.visit(cd);
+		inAggregate--;
 	}
 
-	override void visit(const FunctionBody fb)
+	override void visit(AST.FuncDeclaration fd)
 	{
-		if (fb.specifiedFunctionBody && fb.specifiedFunctionBody.blockStatement !is null)
-			visit(fb.specifiedFunctionBody.blockStatement);
-		else if (fb.shortenedFunctionBody && fb.shortenedFunctionBody.expression !is null)
-			visitReturnExpression(fb.shortenedFunctionBody.expression);
-	}
+		import dmd.astenums : Tbool;
 
-	override void visit(const BlockStatement bs)
-	{
-		if (bs.declarationsAndStatements is null)
-			return;
-		if (bs.declarationsAndStatements.declarationsAndStatements is null)
-			return;
-		if (bs.declarationsAndStatements.declarationsAndStatements.length != 1)
+		if (!inAggregate)
 			return;
-		visit(bs.declarationsAndStatements);
-	}
 
-	override void visit(const ReturnStatement rs)
-	{
-		if (inStruct == 0 || line == size_t.max) // not within a struct yet
+		if (!fd.ident || fd.ident.toString() != "empty")
 			return;
-		visitReturnExpression(rs.expression);
-	}
 
-	void visitReturnExpression(const Expression expression)
-	{
-		if (!expression || expression.items.length != 1)
-			return;
-		UnaryExpression unary = cast(UnaryExpression) expression.items[0];
-		if (unary is null)
-			return;
-		if (unary.primaryExpression is null)
+		AST.TypeFunction tf = fd.type.isTypeFunction();
+
+		if (!tf || !tf.next || !tf.next.ty)
 			return;
-		if (unary.primaryExpression.primary != tok!"false")
+
+		AST.ReturnStatement rs = fd.fbody ? fd.fbody.isReturnStatement() : null;
+
+		if (rs)
+		{
+			AST.IntegerExp ie = cast(AST.IntegerExp) rs.exp;
+
+			if (ie && ie.value == 0)
+				addErrorMessage(cast(ulong) fd.loc.linnum, cast(ulong) fd.loc.charnum, KEY,
+				"Use `enum bool empty = false;` to define an infinite range.");
+		}
+
+		AST.CompoundStatement cs = fd.fbody ? fd.fbody.isCompoundStatement() : null;
+		
+		if (!cs || (*cs.statements).length == 0)
 			return;
-		addErrorMessage(line, column, KEY, MESSAGE);
+
+		if (auto rs1 = (*cs.statements)[0].isReturnStatement())
+		{
+			AST.IntegerExp ie = cast(AST.IntegerExp) rs1.exp;
+
+			if (ie && ie.value == 0)
+				addErrorMessage(cast(ulong) fd.loc.linnum, cast(ulong) fd.loc.charnum, KEY,
+				"Use `enum bool empty = false;` to define an infinite range.");
+		}
+
+		super.visit(fd);
 	}
 
-	override void visit(const Unittest u)
+	override void visit(AST.UnitTestDeclaration ud)
```

Comment:
What's the point of this empty visit method?

Reply:
Well, I guess it's purpose is to not verify the code inside a `unittest` block for this particular check, and I tried to mimic the behavior of the actual D-Scanner here, but I think we can get rid of that empty method. What do you think about that? I can't see why you wouldn't check `unittest` blocks

Reply:
If that's what D-scanner does, let's just keep it as is and fix it later if the need arises.